x/dex/keeper/query/grpc_query_get_latest_price.go

```diff
@@ -0,0 +1,26 @@
+package query
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k KeeperWrapper) GetLatestPrice(goCtx context.Context, req *types.QueryGetLatestPriceRequest) (*types.QueryGetLatestPriceResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	blockTimeStamp := uint64(ctx.BlockTime().Unix())
```

Comment:
@codchen @LCyson Right here is where we get the current block timestamp and use that for the current price. One issue is that we may not have a price written for this, but may exist for the previous block.

Reply:
yeah we should use a iterator to find the closest price, otherwise this would return not found most of the time

Reply:
@codchen check latest commit, it loops through all prices and finds the latest given price